src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -239,15 +247,14 @@ abstract class FixupSession(
   }
 
   fun retry() {
-    // TODO: The actual prompt we sent is displayed as ghost text in the text input field, in VS
-    // Code.
-    // E.g. "Write a brief documentation comment for the selected code <etc.>"
-    // We need to send the prompt along with the lenses, so that the client can display it.
+    val instruction = instruction
+    undo()
```

Comment:
Unfortunately that will collide with my other changes and I'm not sure if it's step in the right direction.

We are very close to not need `undo` on the client side, and do everything in the agent:
https://github.com/sourcegraph/jetbrains/pull/1420
But for it to work properly `retry` should be implemented as the agent command, and it should trigger undo itself first.

Reply:
I understand what you're saying, but consider:

- it's completely broken for users today.
- fixing it properly is a lot of work. At least 2 days, maybe more. I tried already, and backed out, because the VS Code extension implementation of Retry is just completely tangled up with the UI. It will be a full reimplementation with new tests, etc.

So. With that in mind, can we please let this through for now?

Reply:
Sure, let's do it.
I still have one concern regarding potential edge case - calling undo in agent not causes it to send us edit information which is doina a real undo.
In most cases we were just missing/ignoring it, because current session was already disposed when that messages was coming in, and so it was ignored.
I'm not sure if that will be the case in your situation. It may be dependent on timing and only manifest as an issue sometimes I think.

That said let's merge it, and I will give it some testing as well as part of my other work.

Reply:
Ah,, interesting. I've seen that happening quite a bit, and yeah, we do ignore the extra, but that does need to be cleaned up.